wire-ios-sync-engine/Source/UserSession/ZMUserSession+LegalHold.swift

```diff
@@ -42,65 +42,71 @@ extension ZMUserSession {
 
     public func accept(legalHoldRequest: LegalHoldRequest, password: String?, completionHandler: @escaping (_ error: LegalHoldActivationError?) -> Void) {
 
-        func complete(error: LegalHoldActivationError?) {
-            syncManagedObjectContext.saveOrRollback()
-
-            DispatchQueue.main.async {
-                completionHandler(error)
-            }
-        }
-
         guard let apiVersion = BackendInfo.apiVersion else {
-            return complete(error: .missingAPIVersion)
+            return completionHandler(.missingAPIVersion)
         }
 
-        syncManagedObjectContext.performGroupedBlock {
-            let selfUser = ZMUser.selfUser(in: self.syncManagedObjectContext)
+        // 1) Check the state
+        let selfUser = ZMUser.selfUser(in: self.managedObjectContext)
```

Comment:
it's the viewContext here?

Reply:
Yes, `managedObjectContext` is the viewContext in this context (ZMUserSession).